tests/attr/test_stat.py

```diff
@@ -0,0 +1,155 @@
+#!/usr/bin/env python3
+import numpy as np
+import torch
+import random
+
+from captum.attr._utils.stat import (
+    StatGraph,
+    Mean,
+    MSE,
+    Var,
+    StdDev,
+    SampleStdDev,
+    SampleVar,
+    Min,
+    Max,
+    _topo_sort,
+)
+from .helpers.utils import BaseTest, assertTensorAlmostEqual
+
+
+def get_values(n=100, lo=None, hi=None, integers=False):
+    for _ in range(n):
+        if integers:
+            yield random.randint(lo, hi)
+        else:
+            yield random.random() * (hi - lo) + lo
+
+
+class Test(BaseTest):
+    def test_div0(self):
+        stat_graph = StatGraph().add(Var).add(Mean)
+        summ = stat_graph.summary
+        self.assertIsNone(summ["mean"])
+        self.assertIsNone(summ["variance"])
+
+        stat_graph.traverse(torch.tensor(10))
+        summ = stat_graph.summary
+        assertTensorAlmostEqual(self, summ["mean"], 10)
+        self.assertIsNone(summ["variance"])
+
+        stat_graph.traverse(torch.tensor(10))
+        summ = stat_graph.summary
+        assertTensorAlmostEqual(self, summ["mean"], 10)
+        assertTensorAlmostEqual(self, summ["variance"], 0)
+
+    def test_var_defin(self):
+        """ 
+        Variance is avg squared distance to mean. Thus it should be positive. This test 
+        is to ensure this is the case.
+
+        To test it, we will we make a skewed distribution leaning to one end (either very 
+        large or small values).
+
+        We will also compare to numpy and ensure it is approximately the same. This 
+        is assuming numpy is correct, for which it should be.
+        """
+        SMALL_VAL = -10000
+        BIG_VAL = 10000
+        AMOUNT_OF_SMALLS = [100, 10]
+        AMOUNT_OF_BIGS = [10, 100]
+        for sm, big in zip(AMOUNT_OF_SMALLS, AMOUNT_OF_BIGS):
+            graph = StatGraph().add(Var)
+            values = []
+            for i in range(sm):
+                values.append(SMALL_VAL)
+                graph.traverse(torch.tensor(SMALL_VAL, dtype=torch.float64))
+
+            for i in range(big):
+                values.append(BIG_VAL)
+                graph.traverse(torch.tensor(BIG_VAL, dtype=torch.float64))
+
+            actual_var = np.var(values)
+            actual_var = torch.from_numpy(np.array(actual_var))
+
+            var = graph.summary["variance"]
+
+            assertTensorAlmostEqual(self, var, actual_var)
+            self.assertTrue((var > 0).all())
+
+    def test_stats_random_data(self):
+        N = 1000
+        BIG_VAL = 100000
```

Comment:
As `N` or `BIG_VAL` grows, rounding errors occur and causes a difference in numpy; note numpy has it implemented as `mean((x-m)**2)` (i.e. without a rolling variance).

Hence why these values are chosen to be low (in order for the tests to pass), but this is a bit hacky, not sure what else I can do though.